internal/pkg/service/buffer/storage/local/writer/disksync/disksync.go

```diff
@@ -290,3 +263,29 @@ func (s *Syncer) doSync() *notify.Notifier {
 
 	return notifier
 }
+
+func (s *Syncer) syncLoop() {
+	ticker := s.clock.Ticker(s.config.CheckInterval)
+
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		defer ticker.Stop()
+
+		// Periodically check the conditions and start synchronization if any condition is met
+		for {
+			select {
+			case <-s.ctx.Done():
+				// The Close method has been called
+				return
+			case <-ticker.C:
+				countTrigger := s.writeOpsCount.Load() >= uint64(s.config.CountTrigger)
+				bytesTrigger := datasize.ByteSize(s.bytesToSync.Load()) >= s.config.BytesTrigger
+				intervalTrigger := s.clock.Now().Sub(s.lastSyncAt.Load()) >= s.config.IntervalTrigger
+				if countTrigger || bytesTrigger || intervalTrigger {
+					s.TriggerSync(false)
+				}
+			}
+		}
+	}()
+}
```

Comment:
`syncLoop` je uz jedine miesto, kde sa v tomto subore vola `TriggerSync`, je to tak uplne predvidatelne, kazdych `5ms` sa skontroluju triggers.